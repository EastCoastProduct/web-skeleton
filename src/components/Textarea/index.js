import React, { PropTypes } from 'react';

const Textarea = (props) => {
  const { label, input, meta: { active, touched, error }, ...other } = props;

  return (
    <div>
      <label htmlFor={input.name}>
        {label}
        <textarea id={input.name} {...input} {...other} />
      </label>
      {!active && touched && error && <p>{error}</p>}
    </div>
  );
};

Textarea.propTypes = {
  label: PropTypes.string,
  input: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired,
};

export default Textarea;
