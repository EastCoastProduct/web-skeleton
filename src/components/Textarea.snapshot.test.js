import React from 'react';
import renderer from 'react-test-renderer';
import Textarea from './Textarea';

describe('Textarea component snapshot', () => {
  it('renders basic required data', () => {
    const meta = {
      active: false,
      error: false,
      touched: false,
    };
    const input = {
      name: 'email',
      value: '',
    };
    const tree = renderer.create(
      <Textarea meta={meta} input={input} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders label and is focused', () => {
    const meta = {
      active: true,
      error: false,
      touched: true,
    };
    const input = {
      name: 'email',
      value: 'test@email.com',
    };
    const tree = renderer.create(
      <Textarea meta={meta} input={input} label="Email" />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders label and input is required', () => {
    const meta = {
      active: false,
      error: 'Input required',
      touched: true,
    };
    const input = {
      name: 'email',
      value: '',
    };
    const tree = renderer.create(
      <Textarea meta={meta} input={input} label="Email" />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders label and input is invalid', () => {
    const meta = {
      active: false,
      error: 'Input is invalid',
      touched: true,
    };
    const input = {
      name: 'email',
      value: 'notAnEmail',
    };
    const tree = renderer.create(
      <Textarea meta={meta} input={input} label="Email" />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders extra info and it is active and invalid', () => {
    const meta = {
      active: true,
      error: 'Input is invalid',
      touched: true,
    };
    const input = {
      name: 'email',
      value: 'notAnEmail',
    };
    const tree = renderer.create(
      <Textarea
        meta={meta}
        input={input}
        label="Email"
        type="email"
        placeholder="Email"
        maxLength="1000" />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
