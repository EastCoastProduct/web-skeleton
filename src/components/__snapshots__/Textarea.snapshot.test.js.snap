exports[`Textarea component snapshot renders basic required data 1`] = `
<div>
  <label
    htmlFor="email">
    <textarea
      id="email"
      name="email"
      value="" />
  </label>
</div>
`;

exports[`Textarea component snapshot renders extra info and it is active and invalid 1`] = `
<div>
  <label
    htmlFor="email">
    Email
    <textarea
      id="email"
      maxLength="1000"
      name="email"
      placeholder="Email"
      type="email"
      value="notAnEmail" />
  </label>
</div>
`;

exports[`Textarea component snapshot renders label and input is invalid 1`] = `
<div>
  <label
    htmlFor="email">
    Email
    <textarea
      id="email"
      name="email"
      value="notAnEmail" />
  </label>
  <p>
    Input is invalid
  </p>
</div>
`;

exports[`Textarea component snapshot renders label and input is required 1`] = `
<div>
  <label
    htmlFor="email">
    Email
    <textarea
      id="email"
      name="email"
      value="" />
  </label>
  <p>
    Input required
  </p>
</div>
`;

exports[`Textarea component snapshot renders label and is focused 1`] = `
<div>
  <label
    htmlFor="email">
    Email
    <textarea
      id="email"
      name="email"
      value="test@email.com" />
  </label>
</div>
`;
